api/handlers.go

```diff
@@ -165,3 +167,96 @@ func (api *RecipeAPI) AddRecipeHandler(w http.ResponseWriter, req *http.Request)
 	w.Header().Set("content-type", "application/json")
 	w.Write(output)
 }
+
+//UpdateRecipeHandler - update specific recipe by ID
+func (api *RecipeAPI) UpdateRecipeHandler(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+
+	//Get Update Recipe ID
+	vars := mux.Vars(req)
+	recipeID := vars["id"]
+	logD := log.Data{"recipe_id": recipeID}
+
+	// Read body
+	b, err := ioutil.ReadAll(req.Body)
+	defer req.Body.Close()
+	if err != nil {
+		log.Event(ctx, "error in reading request body", log.ERROR, log.Error(err))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// Unmarshal to the shape of Response struct
+	var recipeUpdate recipe.Response
+	err = json.Unmarshal(b, &recipeUpdate)
+	if err != nil {
+		log.Event(ctx, "error returned from json unmarshal", log.ERROR, log.Error(err))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	recipeUpdate.ID = recipeID
+
+	// Use exisiting (old) recipe and fills in the fields of the update recipe with data from exisiting when particular field is nil
+	var oldRecipe *recipe.Response
+	oldRecipe, err = api.dataStore.Backend.GetRecipe(recipeID)
+	if err == errs.ErrRecipeNotFound {
+		log.Event(ctx, "recipe not found in mongo", log.ERROR, log.Error(err), logD)
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+	if err != nil {
+		log.Event(ctx, "error getting recipe from mongo", log.ERROR, log.Error(err), logD)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	recipeUpdate = UpdateRecipeFieldsFromPrevious(recipeUpdate, *oldRecipe)
+
+	//Validation to check if all the recipe fields are entered
+	err = recipeUpdate.Validate()
+	if err != nil {
+		log.Event(ctx, "bad request error as incomplete updated recipe given in request body", log.ERROR, log.Error(err))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Update Recipe to Mongo
+	err = api.dataStore.Backend.UpdateRecipe(recipeID, recipeUpdate)
+	if err != nil {
+		log.Event(ctx, "error updating recipe to mongo", log.ERROR, log.Error(err), logD)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// Marshal to output the newly updated recipe
+	updatedRecipe, _ := api.dataStore.Backend.GetRecipe(recipeID)
+	output, err := json.Marshal(updatedRecipe)
+	if err != nil {
+		log.Event(ctx, "error getting updated recipe from mongo", log.ERROR, log.Error(err), logD)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("content-type", "application/json")
+	w.Write(output)
+}
+
+//UpdateRecipeFieldsFromPrevious - uses the exisiting recipe and fills in the fields of the update recipe with data from exisiting when field is nil
+func UpdateRecipeFieldsFromPrevious(updates recipe.Response, oldRecipe recipe.Response) (updateRecipe recipe.Response) {
+
+	if updates.Alias == "" {
```

Comment:
I would also invert this logic:

```
{
    ...
    updateRecipe := currentRecipe

    if update.Alias != "" { 
        updateRecipe.Alias = update.Alias
    }

    ...

    return updateRecipe
}
```

The same with the other if statements

Reply:
Done in commit "Update different fields of recipe and add endpoint"